dvc/repo/artifacts.py

```diff
@@ -158,10 +158,31 @@ def get_path(self, name: str):
         dirname = m.group("dirname")
         if dirname:
             dirname = dirname.rstrip(SEPARATOR_IN_NAME)
+
+        return dirname, m.group("name")
+
+    def get_path(self, name: str):
+        """Return fspath for the given artifact relative to the git root."""
+        from dvc.fs import GitFileSystem
+
+        dirname, artifact_name = self.parse_path(name)
         dvcyaml = os.path.join(dirname, PROJECT_FILE) if dirname else PROJECT_FILE
-        artifact_name = m.group("name")
+
+        # `name`/`dirname` are expected to be a git root relative.
+        # We convert it to dvc-root relative path so that we can read artifacts
+        # from dvc.yaml file.
+        # But we return dirname intact, as we want to return a git-root relative path.
+        # This is useful when reading from `dvcfs` from remote.
+        assert self.scm
```

Comment:
My understanding is that, using `Artifacts` always require a git repo, is that correct?

Reply:
I guess it's possible someone could do `dvc init --no-scm` and add an artifact, so probably best to fallback to dvc root if it's not that hard, although I can't think of a realistic situation for it, so not that strong an opinion.

Reply:
There are two components here in artifacts:

1. One is `git` tagging.
2. The other is `artifacts` schema in `dvc.yaml`.

From what I see, the way to interact with `artifacts` is via tag name, generated either via `Studio` and/or `gto`.

Both of those require the repo to be a git repo, as tagging is involved.

DVC accepts the tag `name` from the users, resolves it to a git commit, reuses the tag name to find the appropriate `dvc.yaml` and the artifact.

So, I am not sure the artifacts can work without it being a dvc repo.

---

I find the whole `artifacts` convoluted, so please correct me if I'm wrong, and if there are other ways to interact with `artifacts` without using git tags.

Reply:
Yes, correct. I think it is possible to use `artifacts` in `dvc.yaml` without a Git repo, although I agree it's not very useful without a Git repo to do tagging, and I'm not sure there's any scenario where it makes sense to read artifacts without a Git repo, so again not a strong opinion here.